app/controllers/events_controller.rb

```diff
@@ -5,10 +5,14 @@ class EventsController < ApplicationController
   # GET /events
   # GET /events.xml
   def index
-    @events = Event.page(params[:page]).order("date desc")
+    @events = Event.unscoped.page(params[:page]).order("date DESC, start_time ASC")
+    @events = @events.where(:type => params[:type]) if params[:type]
+    @events = @events.where("date IN (?)", params[:dates]) if params[:dates]
 
     respond_to do |format|
       format.html # index.html.erb
+      format.json { render :json => @events.as_json(methods: [:short_title, :attendance_options]).group_by{|e| e["date"] } }
```

Comment:
Line is too long. [124/80]<br>Use the new Ruby 1.9 hash syntax.<br>Space missing to the left of {.<br>Space between { and | missing.